tool/password/src/main/java/com/github/mizool/tool/password/PasswordHashTool.java

```diff
@@ -39,14 +43,18 @@ public static void main(String[] args)
             }
             else
             {
-                PasswordHashTool passwordHashTool = new PasswordHashTool(PasswordHasherRegistry.getDefaultHasher(),
-                    parameters.getPassword());
+                ServiceLoaderInstanceImpl<PasswordHasher>
+                    instances
+                    = new ServiceLoaderInstanceImpl<>(PasswordHasher.class);
```

Comment:
Okay, I now realize that in proposing `ServiceLoaderInstanceImpl`, I blindly assumed the CLI tool would use the `PasswordHasherRegistry` and would therefore need to supply an `Instance`. While that would work, it would mean adding a `getAllHashers()` method to the registry that arguably no other call site would ever need.

The code right now is a mixture of two approaches, and to be honest, I'm undecided between them. Please choose one of the following approaches:

1. Encapsulate instantiation of the `Instance<PasswordHasher>`: add Guice, add a Guice module class that knows how to provide an `Instance<PasswordHasher>`, then have `PasswordHashTool` ask Guice for a ready-made instance of `PasswordHasherRegistry`, then invoke a newly added method `getAllHashers()` on it.
2. Simplify things because we're not using the registry anyway: remove the `ServiceLoaderInstanceImpl`, then have `PasswordHashTool` call `MetaInfServices.instances()`. 
    - Note that this is different from the original commit where the registry interacted with `MetaInfServices`. Now, the registry is usable for both CDI and non-CDI environments and - in the latter case - leaves the application in charge of supplying an implementation of `Instance<PasswordHasher>`. `PasswordHashTool`, on the other hand, is not interested in the registry, so it doesn't need an `Instance<PasswordHasher` and can be safely hard-coded to use `MetaInfServices`.